libvast/src/system/configuration.cpp

```diff
@@ -229,6 +246,11 @@ caf::error configuration::parse(int argc, char** argv) {
   }
   // Flatten everything for simplicity.
   merged_config = flatten(merged_config);
+  // Overwrite current config with VAST_ environment variables, potentially
+  // appending new config keys at the end.
+  for (const auto& [key, value] : detail::environment())
+    if (key.starts_with("VAST_") && !value.empty())
+      merged_config[env_to_config_key(key)] = std::string{value};
```

Comment:
I don't think this handles nested keys correctly, does it?

Reply:
`merged_config` is flattened with dots.

Reply:
In that case it is irrelevant.